common/springOverrides.lua

```diff
@@ -21,9 +21,8 @@ if Spring.GetModOptions then
 		end
 	end
 
-	Spring.GetModOptions = function ()
-		-- Returning the table itself would allow callers to mutate the table
-		-- Copying it ensures each caller gets its own copy
-		return table.copy(modOptions)
+	local modOptionsCopy = table.copy(modOptions)
+	Spring.GetModOptions = function()
+		return modOptionsCopy
```

Comment:
This needs to return a new instance of the table every time it's called, otherwise one caller can mutate the results for other callers.